jobs,webui: define a new endpoint for job statusJobs endpoint

spaskob opened the pull request:
A request to _status/job/{job_id} will
return info about this job id if it is
currently being executed.

Fixes #45016.

Release note (admin ui change): end point /_status/job/{job_id}
will display status info about this job.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45094)
<!-- Reviewable:end -->

Commit:
jobs,webui: define a new endpoint for job status

A request to _status/job/{job_id} will
return info about this job id if it is currently running on
this node.

Fixes #45016.

Release note (admin ui change): end point /_status/job/{job_id}
will display status info about this job.

dt reviewed (approved):
> Reviewed 3 of 4 files at r1, 5 of 5 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45094)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt and @spaskob)

___
*[pkg/jobs/jobs.go, line 523 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/45094#-M0Z4_zm02QEc9qcchkD:-M0Z4_zn18C00Ce6KLBH:b-84jan9) ([raw file](https://github.com/cockroachdb/cockroach/blob/3484ced05a98e6bc4b39e77f7d0b6d73a83c4523/pkg/jobs/jobs.go#L523)):*
> ```Go
> 	return jobspb.Job{
> 		Id:       *j.ID(),
> 		Payload:  &j.mu.payload,
> ```

I think we might get in trouble returning pointers to a lock-protected field when the pointers outlive the holding of the lock -- should these both take copies?


<!-- Sent from Reviewable.io -->

spaskob commented:
bors r+

spaskob commented:
bors r+
